packages/dialog/src/provider/dialogs-provider.ts

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+export type IDialogController = {
+  openDialog: (dialogId: string) => void;
+  closeDialog: (dialogId: string) => void;
+  openedDialogs: string[];
+};
+
+const noOpDialogControl = () => {
+  console.error('No dialog controller implemented');
+};
+
+export const UiDialogsControllerContext = React.createContext<IDialogController>({
+  openDialog: noOpDialogControl,
+  closeDialog: noOpDialogControl,
+  openedDialogs: [],
+});
+
+UiDialogsControllerContext.displayName = 'UiDialogsControllerContext';
```

Comment:
This is the context that will be used for each client to provide a state handling, we could provide a very generic handling from the repo, but as of now its a noOp, which means client have to implement it.